src/SharpCompress/IO/BufferedSubStream.cs

```diff
@@ -20,29 +20,41 @@ internal class BufferedSubStream(Stream stream, long origin, long bytesToRead)
 
     public override void Flush() { }
 
-    public override long Length => BytesLeftToRead;
+    public override long Length => BytesLeftToRead + _cacheLength - _cacheOffset;
 
     public override long Position
     {
         get => throw new NotSupportedException();
         set => throw new NotSupportedException();
     }
 
+    private void RefillCache()
+    {
+        var count = (int)Math.Min(BytesLeftToRead, _cache.Length);
```

Comment:
This is a new bit of logic here; the old code would always request `_cache.Length` bytes when refilling. It probably wasn't hurting anything but I figure it's better to request only what we need.